sssd_test_framework/utils/sssd.py

```diff
@@ -303,19 +310,40 @@ def clear(self, *, db: bool = True, memcache: bool = True, config: bool = False,
 
     def set_service_user(self, user: str) -> None:
         """
-        Set [sssd]/user option.
+        Reconfigures 'sssd.service' systemd service description
+        to run SSSD service under 'user' (only 'root' or 'sssd'
+        are supported by SSSD).
+        Take a note, this currently doesn't handle reconfiguration
+        of socket activated services.
 
         :param user: Option value to set.
         :type user: str
-        :raises ValueError: If required feature wasn't built.
+        :raises ValueError: in case error happens.
         """
-        if isinstance(self.host, BaseHost):
-            if (user != "root") and (not self.host.features["non-privileged"]):
-                raise ValueError("SSSD was built without support of running under non-root")
+        if isinstance(self.host, ClientHost):
+            if not self.host.features["non-privileged"]:
+                return  # service user configuration isn't supported at all
         else:
             raise ValueError("Unexpected host type")
 
-        self.sssd["user"] = user
+        if user == self.host._sssd_service_user:
+            return  # requested service user matches default, nothing to do
+
+        service_file = "/usr/lib/systemd/system/sssd.service"
+        self.fs.backup(service_file)
+        cmd = f'sed -i "s/^User=.*/User={user}/g" {service_file}\n'
+        cmd += f'sed -i "s/^Group=.*/Group={user}/g" {service_file}\n'
+        if user == "root":
+            cmd += f'sed -i "s/^#SupplementaryGroups=sssd$/SupplementaryGroups=sssd/g" {service_file}\n'
+            cmd += f'sed -i "s/sssd:sssd/root:root/g" {service_file}\n'
+        elif user == "sssd":
+            cmd += f'sed -i "s/^SupplementaryGroups=sssd$/#SupplementaryGroups=sssd/g" {service_file}\n'
+            cmd += f'sed -i "s/root:root/sssd:sssd/g" {service_file}\n'
+        else:
+            raise ValueError("Unexpected value of 'user'")
+        cmd += f"chown -f {user}:{user} /var/lib/sss/db/*.ldb || true"
+        self.host.conn.run(cmd)
+        self.host.conn.run("systemctl daemon-reload")
```

Comment:
`self.svc.reload_daemon`

Reply:
I can try again, but I tried this initially and it didn't work :-/

Reply:
Works.